datadog_checks_downloader/tests/test_downloader.py

```diff
@@ -250,10 +224,27 @@ def test_local_wheels_signer_signature_leaf_error(distribution_name, distributio
             http_url,
         ]
 
-        with pytest.raises(NoWorkingMirrorError) as exc:
+        with pytest.raises(UnsignedMetadataError) as exc:
             _do_run_downloader(argv)
 
-    assert "BadSignatureError('wheels-signer-" in str(exc)
+    assert "wheels-signer-a was signed by 0/1 keys" in str(exc)
+
+
+@pytest.mark.offline
```

Comment:
While we are at it, could we also quickly add a test for a bad target (i.e., a target that has been deliberately tampered with at rest compared to the signed metadata)?

Reply:
I've added that test now, which pretty much asked for a more manageable approach to creating "tampered" versions of the baseline repo copy. The gist of it is that the modifications are now made on-the-fly to avoid generating a new zip file with every little change. The changes involved are:

- The script to create the offline copy of the data now only generates a single zip file with the "good" repo.
- The copy with the tampered signature was removed from the test data folder.
- I added a hook to the local mock server that allows injection of a function to modify the "good" baseline repo before starting to serve it, and then modified the bad signature test and added a new test with a tampered target by using that hook.